### PR TITLE
Key bindings

### PR DESCRIPTION
**Pull request by cooper667**
# Description

This allows us to use an external map of keybinding to methods. It's mostly typesafe (and has no errors). I'll do the following TODOs in another PR

Main change is Toolboxes define the methods that can have keybindings in the following way: https://github.com/gliff-ai/annotate/pull/44/files#diff-94f3880be2d1abd06d206719935522d35557921fce534b4b9e79412c68e9ec95R22

and then add a listener when they mount to listen for custom events with these names. This means keypresses (and any other way we choose to trigger these) are abstracted away from the canvas and hadnled in the UI

Still todo:

- Better typing on the methods allowed (import them from the toolboxes)
- Modifier keys
- (Maybe) namespace the events to prevent conflicts

### CODE REVIEW

**Comment by cooper667** [comment]
Ugh where is package-lock.json ! It's in my local. I'll fix that shortly.

**Comment by ChasNelson1990** [comment]
I think namespaces could be great, especially in the context of #37

**Comment by ChasNelson1990** [comment]
> Ugh where is package-lock.json ! It's in my local. I'll fix that shortly.

It seems to be here but has lost 9000 lines lol

File: src/baseCanvas/Canvas.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ export interface Props {
 }
 export class BaseCanvas extends Component<Props> {
   private name: string;
-  private canvas: HTMLCanvasElement;
+  public canvas: HTMLCanvasElement;
```
Comment (ChasNelson1990) [inline src/baseCanvas/Canvas.tsx:38]: What do you need this for? You can actually get a read-only version of the canvas from the canvas context (which is already public) if you just need things like height and width or something?

**Review by ChasNelson1990 — DISMISSED** [review]

File: src/baseCanvas/Canvas.tsx (hunk 1 above)
Comment (cooper667) [reply src/baseCanvas/Canvas.tsx:38]: ah i don't think i do need it now, I'll change it back

**Review by ChasNelson1990 — APPROVED** [review]

**Review by SilviaZeta — APPROVED** [review]

**Pull request merged**